device-types/Juniper/QFX5100-24Q-2P.yaml

```diff
@@ -0,0 +1,99 @@
+---
+manufacturer: Juniper
+model: QFX5100-24Q-2P
```

Comment:
The `-2P` variant is not mentioned anywhere in juniper's documentation - where is this from ?
https://www.juniper.net/content/dam/www/assets/datasheets/us/en/switches/qfx5100-ethernet-switch-datasheet.pdf

Reply:
Not in the product sheet but from device itself. Which one you go with usually?

```
> show chassis hardware
Hardware inventory:
Item             Version  Part number  Serial number     Description
Chassis                                XXXXXXXXXXXX      QFX5100-24Q-2P
Pseudo CB 0
Routing Engine 0          BUILTIN      BUILTIN           QFX Routing Engine
```

Reply:
try with `show chassis hardware models` and `show chassis hardware clei-models` :)
In any case I think we should go with the QFX5100-24Q but I'm waiting for this first.

Reply:
```
> show chassis hardware models
error: command is not valid on the qfx5100-24q-2p
```

```
> show chassis hardware clei-models
Hardware inventory:
Item             Version  Part number  CLEI code         FRU model number
Routing Engine 0          BUILTIN      XXX        QFX5100-24Q-AFO
FPC 0            REV 28   650-049942   XXX        QFX5100-24Q-AFO
  PIC 0                   BUILTIN      XXX        QFX5100-24Q-AFO
  PIC 1          REV 02   611-064275   XXX        EX4600-EM-8F
  PIC 2          REV 02   611-064275   XXX        EX4600-EM-8F
Power Supply 0   REV 03   740-041741   XXX        JPSU-650W-AC-AFO
Power Supply 1   REV 03   740-041741   XXX        JPSU-650W-AC-AFO
Fan Tray 0                                               QFX5100-FANAFO
Fan Tray 1                                               QFX5100-FANAFO
Fan Tray 2                                               QFX5100-FANAFO
Fan Tray 3                                               QFX5100-FANAFO
Fan Tray 4                                               QFX5100-FANAFO
```

Reply:
So there is one other model "QFX5100-48S-3AFO.yaml" that has "AFO" in model name, should I drop AFO?

Reply:
there is only on device that is named after the direction of air/inlet. I guess it makes much more sense to name the device without it.